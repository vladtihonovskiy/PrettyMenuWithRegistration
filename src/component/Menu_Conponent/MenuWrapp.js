import React, { Component } from 'react';
import {

    Link,
} from 'react-router-dom'
import { elastic as Menu } from 'react-burger-menu'
class MenuWrapp extends Component{
    constructor(props,context) {
        super(props, context);
        this.state = {
            sorted: true,
            show:true,
        }
    }
    btnClick=()=>{
        this.setState({
            show:!this.state.show
        })
    }

    render(){
        return (
            <div className="MenuWrap">
                <Menu noOverlay  width={ 200 } pageWrapId={ "page-wrap" } >
                    {this.props.MassiveAllMenuItems.map((item) => {
                            return (
                               <Link to={'/' + item}>{item}</Link>
                            )
                        }
                    )
                    }
                </Menu>
            </div>
        )
    }
}
export default MenuWrapp;