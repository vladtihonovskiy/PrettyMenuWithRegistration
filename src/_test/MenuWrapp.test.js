import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MenuWrapp from '../../src/component/Menu_Conponent/MenuWrapp';
import { Provider } from 'react-redux';
import reducer from '../../src/redusers/index';
import {createStore,applyMiddleware} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../../src/sagas'
const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(sagaMiddleware))
)
sagaMiddleware.run(rootSaga);




describe('MenuWrapp', () => {
    const mockFn = jest.fn();
    it('should be defined', () => {
        expect(MenuWrapp).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(
            <Provider store={store}>
                <MenuWrapp MassiveAllMenuItems={['Home', 'About', 'Contact','Product','Logout'] }/>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
    it('should have a button value', () => {
        const wrapper = shallow(   <Provider store={store}>
            <MenuWrapp MassiveAllMenuItems={['Home', 'About', 'Contact','Product','Logout'] } name='button test' handleClick={mockFn} />
        </Provider>);
        // expect(wrapper.props().name).toEqual('button test');
        // expect(tree.find('.button').node.props.value).toEqual('button test');
    });
    // it('should have a click function', () => {
    //     const tree = shallow(
    //         <SingleLine name='button test' handleClick={mockFn} />
    //     );
    //     tree.simulate('click');
    //     expect(mockFn).toHaveBeenCalled();
    // });
});