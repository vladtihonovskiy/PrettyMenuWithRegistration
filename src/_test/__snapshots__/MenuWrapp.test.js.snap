// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuWrapp should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <MenuWrapp
    MassiveAllMenuItems={
        Array [
            "Home",
            "About",
            "Contact",
            "Product",
            "Logout",
          ]
    }
/>,
  "nodes": Array [
    <MenuWrapp
      MassiveAllMenuItems={
            Array [
                  "Home",
                  "About",
                  "Contact",
                  "Product",
                  "Logout",
                ]
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <MenuWrapp
                MassiveAllMenuItems={
                        Array [
                                "Home",
                                "About",
                                "Contact",
                                "Product",
                                "Logout",
                              ]
                }
        />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <MenuWrapp
            MassiveAllMenuItems={
                        Array [
                                    "Home",
                                    "About",
                                    "Contact",
                                    "Product",
                                    "Logout",
                                  ]
            }
/>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <MenuWrapp
          MassiveAllMenuItems={
                    Array [
                              "Home",
                              "About",
                              "Contact",
                              "Product",
                              "Logout",
                            ]
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <MenuWrapp
          MassiveAllMenuItems={
                    Array [
                              "Home",
                              "About",
                              "Contact",
                              "Product",
                              "Logout",
                            ]
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <MenuWrapp
        MassiveAllMenuItems={
            Array [
                "Home",
                "About",
                "Contact",
                "Product",
                "Logout",
              ]
        }
    />
</Provider>,
}
`;
